lib/globalize/active_record/adapter_dirty.rb

```diff
@@ -42,7 +42,9 @@ def clear_dirty
       end
 
       def _reset_attribute name
-        record.send("#{name}=", record.changed_attributes[name])
+        original_value = record.changed_attributes[name]
+        record.send(:clear_attribute_changes, [name])
+        record.send("#{name}=", original_value)
         record.send(:clear_attribute_changes, [name])
```

Comment:
@moiristo so we definitely need to call this twice?

Reply:
@parndt unfortunately, yes. This first prevents the loop and the second is needed to clear the changes of that assignment. It would not be needed if we could assign using the super assignment method (without dirty tracking). This is actually possible since 2.2 (something like `method("#{name}=").super_method.call(original_value)`. That might be an option when support for < 2.2 is dropped.

Reply:
we definitely don't support < 2.4 since 270f0944292894fe762d154d09e8a456b221e1f4

Reply:
Ah, good to know. I see you updated the gemspec already :) I'll check if I can rewrite this then!

Reply:
Actually it is not as easy as I thought. We don't need to call the super method of the assignment, but somewhere it should only use the super method of `def write` instead. I don't think that can be done. The alternative would be to use some flag to skip dirty tracking, but I don't think that would result in a better solution. Therefore, I think we should stick to the original solution.